Code/Editor/NewLevelDialog.cpp

```diff
@@ -105,10 +105,74 @@ CNewLevelDialog::CNewLevelDialog(QWidget* pParent /*=nullptr*/)
     ReloadLevelFolder();
 }
 
+
+
 CNewLevelDialog::~CNewLevelDialog()
 {
 }
 
+void CNewLevelDialog::InitTemplateListWidget() const
+{
+    ui->listTemplates->clear();
+
+    const QIcon defaultIcon(":/NewLevel/res/Prefab_80.svg");
+    const QSize iconSize(128, 128);
+    const QString fileFilter = {"*.prefab"};
+    QStringList templatePaths =
+    {
+        {QString(AZ::Utils::GetEnginePath().c_str()) + QDir::separator() + "Assets/Editor/Prefabs"}
+    };
+    QStringList allTemplateFiles;
+
+    // Try to read a custom template path from the settings registry.
+    if(const AZ::SettingsRegistryInterface* settingsRegistry = AZ::SettingsRegistry::Get(); settingsRegistry)
+    {
+        AZStd::string customTemplatePath;
+        if(settingsRegistry->GetObject(customTemplatePath, RegistryKey_CustomTemplatePath))
+        {
+            char finalPathBuffer[AZ_MAX_PATH_LEN] = { 0 };
+            if(AZ::IO::FileIOBase::GetInstance()->ResolvePath(customTemplatePath.c_str(), finalPathBuffer, AZ_MAX_PATH_LEN))
+            {
+                templatePaths.push_back(finalPathBuffer);
+            }
```

Comment:
To query a string from the Settings Registry, the [Get](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Settings/SettingsRegistry.h#L279-L284) functionoverload should be used and not GetObject.
The `GetObject` function would try to use the Serialize Context to try the marshal a JSON object into a string.


Also FYI, you can make an `AZ::IO::FixedMaxPath` and then use the [.Native()](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/IO/Path/Path.h#L559-L562) function to get a modifiable reference to the underlying `AZStd::fixed_string` type and pass that in directly to the SettingsRegistry `Get()` overload.
Furthermore the FileIOBase class has a `ResolvePath` overload that can populate an AZ::IO::FixedMaxPath.

```suggestion
        if (AZ::IO::FixedMaxPath customTemplatePath;
            settingsRegistry->Get(customTemplatePath.Native(), RegistryKey_CustomTemplatePath))
        {
            if(AZ::IO::FileIOBase::GetInstance()->ResolvePath(customTemplatePath, customTemplatePath))
            {
                templatePaths.emplace_back(QString::fromUtf8(customTemplatePath.c_str(), int(customTemplatePath.Native().size())));
            }
```